cfme/automate/buttons.py

```diff
@@ -160,32 +315,45 @@ def __init__(self, group=None, text=None,
         self.request = request
 
     def create(self):
-        navigate_to(self, 'Add')
-        fill(button_form, {'btn_text': self.text,
-                           'btn_hvr_text': self.hover})
-        if version.current_version() < "5.5":
-            select = DHTMLSelect("div#button_div")
-        else:
-            select = AngularSelect("button_image")
-        select.select_by_value(2)
-        fill(button_form, {'select_dialog': self.dialog.label if self.dialog is not None else None,
-                           'system_process': self.system,
-                           'request': self.request})
-        sel.click(button_form.add_button)
-        flash.assert_success_message('Button "{}" was added'.format(self.hover))
+        view = navigate_to(self, 'Add')
+        view.fill({
+            'text': self.text,
+            'hover': self.hover,
+            'dialog': self.dialog,
+            'system': self.system,
+            'request': self.request,
+            'image': 'Button Image 1'
+        })
+        view.add.click()
+        view = self.create_view(ButtonGroupDetailView, self.group)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_message('Button "{}" was added'.format(self.hover))
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        edited_hover = updates.get('hover', None)
-        fill(button_form, {'btn_hvr_text': edited_hover})
-        sel.click(button_form.save_button)
-        flash.assert_success_message('Button "{}" was saved'.format(edited_hover))
-
-    def delete(self):
-        navigate_to(self, 'Details')
-        cfg_btn("Remove this Button", invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_success_message('Button "{}": Delete successful'.format(self.hover))
+        view = navigate_to(self, 'Edit')
+        changed = view.fill(updates)
+        if changed:
+            view.save.click()
+        else:
+            view.cancel.click()
+        view = self.create_view(ButtonDetailView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        if changed:
+            view.flash.assert_message(
+                'Button "{}" was saved'.format(updates.get('hover', self.hover)))
+        else:
+            view.flash.assert_message(
+                'Edit of Button "{}" was cancelled by the user'.format(self.text))
+
+    def delete(self, cancel=False):
+        view = navigate_to(self, 'Details')
+        view.configuration.item_select('Remove this Button', handle_alert=not cancel)
```

Comment:
Ah right, yes, well it is not used yet in such way so there still is space for fixing that up.